### PR TITLE
GREEN-1126 switch to native go library for xml signing

### PR DESCRIPTION
**Pull request by ebeesonjanrain**

### CODE REVIEW

File: response.go [hunk 1]
```diff
@@ -253,14 +144,14 @@ func NewResponse() *Response {
...
-// AddAttribute add attribute to the Response
+// AddAttribute adds an attribute to the response assertion
 func (resp *Response) AddAttribute(name, value string) {
 	resp.Assertion.AttributeStatement.Attributes = append(resp.Assertion.AttributeStatement.Attributes, Attribute{
 		XMLName: xml.Name{
 			Local: "saml:Attribute",
 		},
 		Name:       name,
-		NameFormat: "urn:oasis:names:tc:SAML:2.0:attrname-format:basic",
+		NameFormat: "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified",
```
Comment (Jared-Janrain) [inline response.go]: Briefly reading about this, it seems like this is more accurate.  Does this change our response in a way that receiving libraries would ever have an issue with?

Comment (ebeesonjanrain) [reply response.go]: no - this is so it will match the existing service's nameformat

File: util/crypto.go [hunk 2]
```diff
@@ -1,4 +1,13 @@
------BEGIN RSA PRIVATE KEY-----
+package util
```
Comment (rkonell-janrain) [inline util/crypto.go]: should this be a test package since it only exposes test functionality?

Comment (ebeesonjanrain) [reply util/crypto.go]: can't import from a *_test package, but I could put it in a separate normal package - maybe call it "testutil"?

**Pull request closed without merging**